src/test/java/ch/uzh/ifi/hase/soprafs21/security/JwtTokenAuthenticationFilterTest.java

```diff
@@ -0,0 +1,129 @@
+package ch.uzh.ifi.hase.soprafs21.security;
+
+import ch.uzh.ifi.hase.soprafs21.entity.User;
+import ch.uzh.ifi.hase.soprafs21.rest.dto.OnlineStatusDto;
+import ch.uzh.ifi.hase.soprafs21.rest.dto.UserOverviewDto;
+import ch.uzh.ifi.hase.soprafs21.security.config.SecurityConstants;
+import ch.uzh.ifi.hase.soprafs21.service.JwtTokenUtil;
+import ch.uzh.ifi.hase.soprafs21.service.UserService;
+import io.jsonwebtoken.Claims;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Optional;
+
+import static org.mockito.Mockito.*;
+
+@SpringBootTest
```

Comment:
@SpringBootTest indicates that the application will be started (normally for the integration tests), but then most of the logic is mocked. I guess it is unit test. 
Maybe one integration test would have been better for this part.

Reply:
added integration test